assets/js-focal-image.js

```diff
@@ -33,19 +42,63 @@ const handleFocalImages = () => {
     return `${clampedPercentage}%`;
   }
 
-  focalImages.forEach(image => {
+  const processFocalImage = (image) => {
+    if (image.dataset.focalProcessed === 'true') return;
+
     const focalX = image.getAttribute('data-focal-x');
     const focalY = image.getAttribute('data-focal-y');
 
     if (focalX !== null && focalY !== null) {
-      const objectPositionX = convertCoordinateToPercentage(focalX);
-      const objectPositionY = convertCoordinateToPercentage(focalY);
+      const positionKey = `${focalX}_${focalY}`;
+
+      let position = computedPositions.get(positionKey);
+
+      if (!position) {
+        const objectPositionX = convertCoordinateToPercentage(focalX);
+        const objectPositionY = convertCoordinateToPercentage(focalY);
+        position = `${objectPositionX} ${objectPositionY}`;
+
+        computedPositions.set(positionKey, position);
+      }
```

Comment:
what about extracting a function here so we can just do `const position = calculatePosition(focalX, focalY)`? Then that function does the caching using that Map